File: livingrimoire start here/LivinGrimoire java/LivinGrimoire/Algorithm.java
package LivinGrimoire;

import java.util.ArrayList;

// a step-by-step plan to achieve a goal
public class Algorithm {
    private ArrayList<Mutatable> algParts = new ArrayList<>();

    public Algorithm(ArrayList<Mutatable> algParts) {
        super();
        this.algParts = algParts;
    }
    public ArrayList<Mutatable> getAlgParts() {
        return algParts;
    }

    public int getSize() {
        return algParts.size();
    }
}

File: livingrimoire start here/LivinGrimoire Kotlin/kotlin/livinGrimoire/Algorithm.kt
package livinGrimoire

// a step-by-step plan to achieve a goal
class Algorithm(algParts: ArrayList<Mutatable>) {
    var algParts = ArrayList<Mutatable>()

    init {
        this.algParts = algParts
    }

    val size: Int
        get() = algParts.size
}